e2e/cypress/integration/messaging/file_upload_in_center_channel_spec.js

```diff
@@ -9,9 +9,11 @@
 
 function waitForImageUpload() {
     // * Verify that the image exists in the post message footer
-    cy.get('#postCreateFooter').should('be.visible').find('div.post-image__column').
-        should('exist').
-        and('be.visible');
+    cy.waitUntil(() => cy.get('#postCreateFooter').within(() => {
+        return cy.get('div.post-image.normal').then((image) => {
+            return Boolean(image);
+        });
+    }));
```

Comment:
This is not really doing retries, to wait for element. To see if it retries/waiting, add logs inside the block and change selector to non-existent element. I recommend to change it into:
```javascript
cy.waitUntil(() => cy.get('#postCreateFooter').then((el) => {
    return el.find('.post-image.normal').length > 0;
}));
```

Reply:
@saturninoabril It works for me. This is how I added the logs
```
    cy.waitUntil(() => cy.get('#postCreateFooter').within(() => {
        return cy.get('div.post-image.normal').then((image) => {
            cy.log('POST IMAGE: ' + Boolean(image)).then(() => {
                return Boolean(image);
            });
        });
    }));
```

I tried both `div.post-image.normal` and `div.post-image1.normal` to see the difference and the latter failed because it doesn't exist (see screenshots)

![Screen Shot 2020-02-13 at 8 27 32 AM](https://user-images.githubusercontent.com/487991/74455879-d8c3c600-4e3a-11ea-8e4d-043b6cf9408d.png)
![Screen Shot 2020-02-13 at 8 28 18 AM](https://user-images.githubusercontent.com/487991/74455892-dd887a00-4e3a-11ea-89e5-ae0687a121b8.png)

Reply:
I guess what you're saying is because it's using get instead of find that the wait is inherent to get timeout and not on the retry to find. Is that what you meant?

Reply:
> ... the latter failed because it doesn't exist (see screenshots)

Yes, it's expected to fail but it should wait (or retry) until it reached the timeout (default 5s). In your case, it failed only on first run (meaning no actual retries happened).